tests/errors/w_cpp_in_c_mode.pyx

```diff
@@ -0,0 +1,10 @@
+# mode: compile
+# tag: warnings, c, no-cpp
+
+cdef extern from *:
+    cdef cppclass X:
+        pass
+
+_WARNINGS="""
+5:9: Using 'cppclass' while Cython is not in C++ mode.
```

Comment:
I'm only testing for one of the warnings since the other was was hard to get as part of the test suite since it picks its own filename.

Reply:
There are another checks for filename not tested anywhere:

https://github.com/cython/cython/blob/50fb5cf1c220c0ecf9440d56612ffc8948e8e22e/Cython/Compiler/Main.py#L168-L170

https://github.com/cython/cython/blob/50fb5cf1c220c0ecf9440d56612ffc8948e8e22e/Cython/Compiler/Main.py#L520-L523

Maybe we can move those checks to separate functions and test them separately in `Cython/Compiler/Tests`. But this can be done definitely in separate PR. Hence I am 👍  for this PR.

Reply:
Good spot. I'll have a think this